include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -224,6 +224,7 @@ struct __destroy_fn
     void
     operator()(_ReferenceType __val) const
     {
+        static_assert(std::is_reference_v<_ReferenceType>);
```

Comment:
I don't understand why this is in this PR.  It is already large, lets limit the scope to what is important for this PR.